packages/overlay/src/Overlay.ts

```diff
@@ -0,0 +1,156 @@
+        const overlay = new Overlay();
+        overlay.dispose = () => {
+            overlay.addEventListener('sp-closed', () => {
+                if (!restored) {
+                    restoreContent();
+                    restored = true;
+                }
+                requestAnimationFrame(() => {
+                    overlay.remove();
+                });
+            });
+            overlay.open = false;
+            overlay.dispose = noop;
+        };
+        let restored = false;
+        const restoreContent = reparentChildren([overlayContent], overlay, {
+            position: 'beforeend',
+            prepareCallback: (el) => {
+                const slot = el.slot;
+                el.removeAttribute('slot');
+                return () => {
+                    el.slot = slot;
+                };
+            },
+        });
+        if (v2) {
+            const options = interactionOrOptions as OverlayOptions;
+            overlay.append(overlayContent);
+            overlay.receivesFocus = options.receivesFocus ?? 'auto';
+            overlay.triggerElement = options.trigger || null;
+            overlay.type = options.type || 'modal';
+            overlay.offset = options.offset || 6;
+            overlay.placement = options.placement;
+            overlay.willPreventClose = !!options.notImmediatelyClosable;
+            requestAnimationFrame(() => {
+                requestAnimationFrame(() => {
+                    // Do we want to "open" this path, or leave that to the consumer?
+                    overlay.open = true;
+                });
+            });
+            return overlay;
+        } else if (overlayContent && options) {
+            const target = targetOrContent;
+            const interaction = interactionOrOptions;
+            overlay.receivesFocus = options.receivesFocus ?? 'auto';
+            overlay.triggerElement = options.virtualTrigger || target;
+            overlay.type =
+                interaction === 'modal'
+                    ? 'modal'
+                    : interaction === 'hover'
+                    ? 'hint'
+                    : 'auto';
+            overlay.offset = options.offset ?? 6;
+            overlay.placement = options.placement;
+            overlay.willPreventClose = !!options.notImmediatelyClosable;
+            // This is super dirty...find a better way.
+            // Maybe imperative open should go _at the end_ of everything?
+            // Having an option is likely useful.
+            // Make imperative overlays less useful?
+            // Delete the imperative approach to an overlay?
+            // Possibly the giving all of the responsiblities to the user is the best path.
+            const parent = target.getRootNode() as Document;
+            if (parent === document) {
+                target.insertAdjacentElement('afterend', overlay);
+            } else {
+                parent.append(overlay);
+            }
```

Comment:
Using the root (either body or the shadow root) minimizes the chance of stray CSS impacting overlay content, and of our injected elements breaking DOM requirements for the user. But we should note that in *imperative* API usage, overlays are added to the *bottom* of their document's root, which is not naturally the next tab stop after their trigger... so it's on the user to ensure accessible tab behavior.

```suggestion
            const parent = target.getRootNode() as Document;
            const root = parent === document ? document.body : parent;
            root.append(overlay);
```

Reply:
I take this back, for right now, as it caused test to fail. The following fits better to the tests that have already been written, and to the expected tab order, even if it is more likely to cause issues as it interacts with CSS rules.
```ts
target.insertAdjacentElement('afterend', overlay);
```

We're just about ready for a new RC and then we can test this new assumption against consuming application before locking it down.